python/paddle/optimizer/nadam.py

```diff
@@ -0,0 +1,337 @@
+            It should be a float number or a 0-D Tensor with shape [] and data type as float32.
+            The default value is 0.999.
+        epsilon (float): A small float value for numerical stability.
+            The default value is 1e-08.
+        weight_decay (float|Tensor, optional): The weight decay coefficient, it can be float or Tensor. The default value is 0.01.
+        momentum_decay (float): momentum momentum_decay. The default value is 0.004.
+        grad_clip (GradientClipBase, optional): Gradient clipping strategy, it's an instance of
+            some derived class of ``GradientClipBase`` . There are three clipping strategies
+            ( :ref:`api_paddle_nn_ClipGradByGlobalNorm` , :ref:`api_paddle_nn_ClipGradByNorm` ,
+            :ref:`api_paddle_nn_ClipGradByValue` ). Default None, meaning there is no gradient clipping.
+        name (str, optional): Normally there is no need for user to set this property.
+            For more information, please refer to :ref:`api_guide_Name`.
+            The default value is None.
+
+    **Notes**:
+        **Currently, NAdam doesn't support sparse parameter optimization.**
+
+    Examples:
+        .. code-block:: python
+
+            >>> import paddle
+
+            >>> inp = paddle.rand([10,10], dtype="float32")
+            >>> linear = paddle.nn.Linear(10, 10)
+            >>> out = linear(inp)
+            >>> loss = paddle.mean(out)
+
+            >>> nadam = paddle.optimizer.NAdam(learning_rate=0.1,
+            ...                     parameters=linear.parameters())
+            >>> out.backward()
+            >>> nadam.step()
+            >>> nadam.clear_grad()
+
+            >>> # Note that the learning_rate of linear_2 is 0.01.
+            >>> linear_1 = paddle.nn.Linear(10, 10)
+            >>> linear_2 = paddle.nn.Linear(10, 10)
+            >>> inp = paddle.uniform(shape=[10, 10], min=-0.1, max=0.1)
+            >>> out = linear_1(inp)
+            >>> out = linear_2(out)
+            >>> loss = paddle.mean(out)
+            >>> opt = paddle.optimizer.NAdam(
+            ...     learning_rate=0.1,
+            ...     parameters=[{
+            ...         'params': linear_1.parameters()
+            ...     }, {
+            ...         'params': linear_2.parameters(),
+            ...         'weight_decay': 0.001,
+            ...         'learning_rate': 0.1,
+            ...         'beta1': 0.8
+            ...     }],
+            ...     weight_decay=0.01,
+            ...     beta1=0.9
+            ... )
+            >>> loss.backward()
+            >>> opt.step()
+            >>> opt.clear_grad()
+
+    """
+
+    _momentum_decay_pow_acc_str = "momentum_decay_pow"
+    _beta2_pow_acc_str = "beta2_pow"
+    _mu_product_acc_str = "mu_product"
+    _moment1_acc_str = "moment1"
+    _moment2_acc_str = "moment2"
+
+    def __init__(
+        self,
+        learning_rate=0.001,
```

Comment:
The default value of `learning_rate` is [0.002](https://github.com/pytorch/pytorch/blob/main/torch/optim/nadam.py#L27) in other deep learning frameworks, Which is better?

Reply:
`learning_rate` of `NAdam` is:

- PyTorch: 0.002
- TensorFlow/Keras: [0.001](https://github.com/keras-team/keras/blob/f6c4ac55692c132cd16211f4877fac6dbeead749/keras/src/optimizers/nadam.py#L41)

I can not tell which is better 😅 It is better for users to set by themselves?

Reply:
Does paper have any suggestions? Which default value is commonly used?

Reply:
The empirical value is 0.002, it is also used for Adam:

> The best learning rate found for SGD was .2, for momentum/NAG was .5, for RMSProp was .001, and for Adam/Nadam was .002. - INCORPORATING NESTEROV MOMENTUM INTO A DAM

but the default value is 0.001 of `paddle.optimizer.Adam(learning_rate=0.001 ...)`.

Let me change it to 0.002 in NAdam?

Reply:
> but the default value is 0.001 of `paddle.optimizer.Adam(learning_rate=0.001 ...)`.
Modifying the default values of existing APIs `Adam` requires more testing and can be omitted from this PR
 
> Let me change it to 0.002 in NAdam?
OK, if there is authoritative evidence, this can be done